client/transport/build.gradle

```diff
@@ -55,3 +55,15 @@ namingConventions {
   //we don't have integration tests
   skipIntegTestInDisguise = true
 }
+
+
+javadoc {
+  dependsOn ':core:javadoc'
+  String host;
+  if (version.endsWith("-SNAPSHOT")) {
+    host = "https://snapshots.elastic.co";
+  } else {
+    host = "https://artifacts.elastic.co";
+  }
+  options.linksOffline host + "/javadoc/org/elasticsearch/elasticsearch/${version}", "../../core/build/docs/javadoc/"
```

Comment:
I think this could be done generically. So inside BuildPlugin, have an iteration over all the projects that depend on the current project, add the javadoc dependsOn, and the linksOffline then for the locally build javadocs. Then any gradle project that eg depends on core automatically can link to its javadocs.